lib/Targets/AIETargetIPU.cpp

```diff
@@ -15,71 +15,76 @@
 #include "mlir/Dialect/Func/IR/FuncOps.h"
 #include "mlir/Tools/mlir-translate/MlirTranslateMain.h"
 
+#include "llvm/ADT/ArrayRef.h"
 #include "llvm/ADT/TypeSwitch.h"
 #include "llvm/Support/Format.h"
+#include <vector>
 
 using namespace mlir;
 using namespace xilinx;
 using namespace xilinx::AIE;
 using namespace xilinx::AIEX;
 
-static void emitProlog(llvm::raw_ostream &os) {
-  os << "00000011\n";
-  os << "01000405\n";
-  os << "01000100\n";
-  os << "0B590100\n";
-  os << "000055FF\n";
-  os << "00000001\n";
-  os << "00000010\n";
-  os << "314E5A5F\n";
-  os << "635F5F31\n";
-  os << "676E696C\n";
-  os << "39354E5F\n";
-  os << "6E693131\n";
-  os << "5F727473\n";
-  os << "64726F77\n";
-  os << "00004573\n";
-  os << "07BD9630\n";
-  os << "000055FF\n";
+namespace {
+
+std::vector<uint32_t> getProlog() {
```

Comment:
Just leaving this as a comment. I'd like to be able to remove this, but haven't had time to dig into it. @STFleming was also looking at removing it

Reply:
Yes just this prolog bit. I am not sure it is necessary but there's some testing to do here, it won't work if it is just deleted.